progenitor/Cargo.toml

```diff
@@ -5,17 +5,22 @@ edition = "2021"
 license = "MPL-2.0"
 repository = "https://github.com/oxidecomputer/progenitor.git"
 description = "An OpenAPI client generator"
+build = "build.rs"
 
 [dependencies]
 progenitor-client = { version = "0.2.1-dev", path = "../progenitor-client" }
 progenitor-impl = { version = "0.2.1-dev", path = "../progenitor-impl" }
 progenitor-macro = { version = "0.2.1-dev", path = "../progenitor-macro" }
 anyhow = "1.0"
 openapiv3 = "1.0.0"
+quote = "1.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 clap = { version = "4.0.29", features = ["derive"] }
 
+[build-dependencies]
+built = { version = "0.5", features = ["git2"] }
```

Comment:
We can remove the feature `git2` if we remove the use of `GIT_VERSION`, which is lifted directly from the docs of https://crates.io/crates/built

Reply:
why is `built` necessary? Is direct use of `evn!("CARGO_PKG_VERSION_PRE")` invalid?